test/__init__.py

```diff
@@ -292,19 +309,22 @@ class ModuleStash:
     modules
     """
 
-    def __init__(self, namespace, modules=sys.modules):
+    def __init__(
+        self, namespace: str, modules: Dict[str, ModuleType] = sys.modules
+    ) -> None:
         self.namespace = namespace
         self.modules = modules
-        self._data = {}
+        self._data: Dict[str, ModuleType] = {}
 
-    def stash(self):
-        self._data[self.namespace] = self.modules.pop(self.namespace, None)
+    def stash(self) -> None:
```

Comment:
We're losing the safety of a KeyError here, what was causing the previous logic to fail?

Reply:
```
test/__init__.py:322: error: Argument 2 to "pop" of "MutableMapping" has incompatible type "None"; expected Module  [arg-type]
```
Shall I revert this and add `type: ignore`?

Reply:
Yeah I think we can add a type: ignore for this instance, thanks for clarifying :)

Reply:
Actually nevermind, this is fine as is! I was reviewing on mobile so didn't get full context.